src/tests/Microsoft.PowerFx.Interpreter.Tests/UserDefinedTests.cs

```diff
@@ -24,5 +25,16 @@ public void NamedFormulaEntryTest(string script, string expression, double expec
             var result = (NumberValue)check.GetEvaluator().Eval();
             Assert.Equal(expected, result.Value);
         }
+
+        [Theory]
+        [InlineData("test2(b: Boolean): Boolean = { Set(a, b); };")]
+        [InlineData("test2(b: Boolean): Boolean = { Set(a, b); Collect(abc, { bcd: 1 }) };")]
+        [InlineData("test2(b: Boolean): Boolean = { Set(a, b); Collect(abc, { bcd: 1 }) }; num = 3;")]
+        public void UserDefinedFunctions(string script)
+        {
+            var result = UserDefinitions.Process(script, null);
+
+            Assert.False(result.HasErrors);
+        }
     }
```

Comment:
>} [](http://example.com/codeflow?start=4&length=1)

We should have negative tests (i.e., unclosed curly braces), tests with errors inside braces (we should still parse other definitions)